File: medium/Q6/pangram.js
// Function to check if a string is pangram
const isPangram = (string) =>
  /* {String} case is lowered and characters are globally replaced by '' other than alphabets in {string} and then converted to Set to get only unique value of characters. Then the size of character is checked. */
  new Set(string
    .toLowerCase()
    .replace(/[^a-z]/g, "")
    ).size === 26; // There are only 26 alphabets in English language.

console.log(isPangram("A quick brown fox jumps over the lazy dog.")); // true
console.log(isPangram("A quick brown fox jumps over the lazy cat.")); // false

File: medium/Q6/pangram.py
import string


def pangram(string_value):
    """ Function to check if a string is a pangram """

    """ We check if it's greater than all letters in lowercase because the the string {string_value} may contain spaces, period(.) and also punctuation """
    return set(string_value.lower()) >= set(string.ascii_lowercase)


if __name__ == "__main__":
    print(pangram("The quick brown fox jumps over the lazy dog."))  # True
    print(pangram("The quick brown fox jumps over the lazy cat."))  # False
